### DISCUSSION TITLE
Dynamically transform header

### DISCUSSION THREAD

**Discussion by zulander1** [dotnet/yarp#602] [Q&A] [answered]
Is there a way to dynamically transform the header based on the request

**Comment by Tratcher** [comment]
There's a whole [Transforms ](https://microsoft.github.io/reverse-proxy/articles/transforms.html)infrastructure for that, though we're still working to make it more extensible (#60).

Can you give an example of the transformation you want to do?

**Comment by zulander1** [comment]
We have a scenario where we have a tomcat application that doesn't handle user management... so we have to create proxy that look's up the user and passes on the role to the tomcat application as a header ...

**Comment by Tratcher** [comment, marked as answer]
Check out https://github.com/microsoft/reverse-proxy/discussions/375 for a similar discussion.
